ui/ducks/confirm-transaction/confirm-transaction.duck.js

```diff
@@ -118,8 +130,32 @@ export default function reducer(state = initState, action = {}) {
         ...state,
         nonce: action.payload,
       };
-    case CLEAR_CONFIRM_TRANSACTION:
-      return initState;
+    case SET_DETAILS_FOR_CONFIRM_TX: {
+      return {
+        ...state,
+        sendTxDetailPerId: {
+          ...state.sendTxDetailPerId,
+          [action.payload.transactionId]: {
+            amountMode: action.payload.amountMode,
```

Comment:
Rather than giving this duck knowledge of these two properties, could we instead just a `boolean` and let the dispatcher simply enable or disable max mode for the a specific transaction?

Reply:
Done in https://github.com/MetaMask/metamask-extension/commit/f362be605aa5deda59e74b27bfe810a05b60db8e